sentinel-transport/sentinel-transport-common/src/main/java/com/alibaba/csp/sentinel/transport/config/TransportConfig.java

```diff
@@ -40,6 +41,8 @@
 
     private static int runtimePort = -1;
 
+	private static String protocol = "http";
```

Comment:
> I suggest using an enum type to make it easy to compare.
> And the protocol should not be a global state but be a kind of state in the addr-port pairs.

In fact, I've thought about creating a new class similar to Tuple2, but it can have three elements. Do you mean it's still stored in Tuple2 in the form of Tuple2.of(protocol + host, port)?

Reply:
> > I suggest using an enum type to make it easy to compare.
> > And the protocol should not be a global state but be a kind of state in the addr-port pairs.
> 
> In fact, I've thought about creating a new class similar to Tuple2, but it can have three elements. Do you mean it's still stored in Tuple2 in the form of Tuple2.of(protocol + host, port)?

Elements grow a new type like Endpoint can be introduced so that the Tuple can be replaced. A Tuple3 type maybe looks more complicated now.

Reply:
> > > I suggest using an enum type to make it easy to compare.
> > > And the protocol should not be a global state but be a kind of state in the addr-port pairs.
> > 
> > 
> > In fact, I've thought about creating a new class similar to Tuple2, but it can have three elements. Do you mean it's still stored in Tuple2 in the form of Tuple2.of(protocol + host, port)?
> 
> Elements grow a new type like Endpoint can be introduced so that the Tuple can be replaced. A Tuple3 type maybe looks more complicated now.

As far as I know, Java can't implement tuples of different lengths in a class unless it's identified by length in a class. I don't know if it's right. If you agree, I'll implement a simple tuple3 as a temporary solution.

Reply:
> > > > I suggest using an enum type to make it easy to compare.
> > > > And the protocol should not be a global state but be a kind of state in the addr-port pairs.
> > > 
> > > 
> > > In fact, I've thought about creating a new class similar to Tuple2, but it can have three elements. Do you mean it's still stored in Tuple2 in the form of Tuple2.of(protocol + host, port)?
> > 
> > 
> > Elements grow a new type like Endpoint can be introduced so that the Tuple can be replaced. A Tuple3 type maybe looks more complicated now.
> 
> As far as I know, Java can't implement tuples of different lengths in a class unless it's identified by length in a class. I don't know if it's right. If you agree, I'll implement a simple tuple3 as a temporary solution.

To be clear, I would show it by examples to make sure we are on the same road:

The old approach:

Tuple2<addr, port>   -> fixed to HTTP protocol

1st approach to improve it:

Tuple3<protocol, addr, port> -> A new type Tuple3 is introduced.

2nd approach:

Endpoint -> A new type Endpoint is introduced as the details below:

```java
public class Endpoint {
    private PROTOCOL protocol;
    private String address;
    private int port;
    // getters and setters
}
```

Reply:
> > > > > I suggest using an enum type to make it easy to compare.
> > > > > And the protocol should not be a global state but be a kind of state in the addr-port pairs.
> > > > 
> > > > 
> > > > In fact, I've thought about creating a new class similar to Tuple2, but it can have three elements. Do you mean it's still stored in Tuple2 in the form of Tuple2.of(protocol + host, port)?
> > > 
> > > 
> > > Elements grow a new type like Endpoint can be introduced so that the Tuple can be replaced. A Tuple3 type maybe looks more complicated now.
> > 
> > 
> > As far as I know, Java can't implement tuples of different lengths in a class unless it's identified by length in a class. I don't know if it's right. If you agree, I'll implement a simple tuple3 as a temporary solution.
> 
> To be clear, I would show it by examples to make sure we are on the same road:
> 
> The old approach:
> 
> Tuple2<addr, port> -> fixed to HTTP protocol
> 
> 1st approach to improve it:
> 
> Tuple3<protocol, addr, port> -> A new type Tuple3 is introduced.
> 
> 2nd approach:
> 
> Endpoint -> A new type Endpoint is introduced as the details below:
> 
> ```java
> public class Endpoint {
>     private PROTOCOL protocol;
>     private String address;
>     private int port;
>     // getters and setters
> }
> ```

I prefer the second approach.

Reply:
> 
> I prefer the second approach.

Yes, I think we have reached the same point now. And another suggestion on the design of arguments of different implementations is based on the decision here.
And I totally agree with you here.😀